src/eve/concepts.py

```diff
@@ -116,11 +115,10 @@ class Config:
 _EVE_NODE_INTERNAL_SUFFIX = "__"
 _EVE_NODE_IMPL_SUFFIX = "_"
 
-AnyNode = TypeVar("AnyNode", bound="BaseNode")
```

Comment:
I'm not sure if we should delete the `AnyNode` TypeVar definition, since it has a different meaning than `BaseNode`. Example:
```
def f1(a: BaseNode) -> BaseNode:
    ...

def f2(a: AnyNode) -> AnyNode:
    ...
```
`f2` always returns the same BaseNode subclass it got as an argument, while `f1` may return a different subclass. Maybe it could be renamed to `NodeT` to make clear that is a TypeVar.